src/ServiceControl/Operations/AuditImporter.cs

```diff
@@ -104,24 +105,27 @@ public async Task<ProcessedMessage> ConvertToSaveMessage(MessageContext message)
                 messageId = DeterministicGuid.MakeId(message.MessageId).ToString();
             }
 
-            var metadata = new Dictionary<string, object>
+            var metadata = new ConcurrentDictionary<string, object>
             {
                 ["MessageId"] = messageId,
                 ["MessageIntent"] = message.Headers.MessageIntent(),
                 ["HeadersForSearching"] = string.Join(" ", message.Headers.Values)
             };
 
-            // TODO: Fan out?
+            var enricherTasks = new List<Task>(enrichers.Length);
```

Comment:
Is there a benefit to doing it this way instead of 

```CSharp
var enricherTasks = enrichers.Select(enricher => enricher.Enrich(message.Headers, metadata));

await Task.WhenAll(enricherTasks).ConfigureAwait(false);
```

Reply:
Also, don't we need to add `ConfigureAwait(false)` after calling `enricher.Enrich(...)`?

Reply:
Yes. One doesn't use LINQ which is good in terms of perf. Plus the LINQ version materializes on WhenAll while the list version materializes and thus schedules the tasks for execution when they are added

A task not awaited doesn't need to be configured with `ConfigureAwait`